src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -22,38 +22,46 @@ public sealed class TempFileCache : IDisposable, ILowMemoryHandler
         private readonly ConcurrentQueue<MemoryStream> _ms = new ConcurrentQueue<MemoryStream>();
 
         private const long MaxFileSizeToKeepInBytes = 16 * Constants.Size.Megabyte;
-        private const int MaxFilesToKeepInCache = 32;
+        internal const int MaxFilesToKeepInCache = 32;
         private int _memoryStreamCapacity = 128 * Constants.Size.Kilobyte;
+        internal const string FilePrefix = "lucene-";
+
+        public int FilesCount => _files.Count;
 
         public TempFileCache(StorageEnvironmentOptions options)
         {
             _options = options;
             string path = _options.TempPath.FullPath;
             if (Directory.Exists(path) == false)
                 Directory.CreateDirectory(path);
-            foreach (string file in Directory.GetDirectories(path, "lucene-*" + StorageEnvironmentOptions.DirectoryStorageEnvironmentOptions.TempFileExtension))
+
+            foreach (string file in Directory.GetFiles(path, $"{FilePrefix}*" + StorageEnvironmentOptions.DirectoryStorageEnvironmentOptions.TempFileExtension))
             {
                 var info = new FileInfo(file);
-                if (info.Length > MaxFileSizeToKeepInBytes ||
-                    _files.Count >= MaxFilesToKeepInCache)
+                if (info.Length > MaxFileSizeToKeepInBytes || _files.Count >= MaxFilesToKeepInCache)
                 {
-                    File.Delete(file);
+                    try
+                    {
+                        File.Delete(file);
```

Comment:
We have `IOExtensions.DeleteFile` for that purpose. It ignores `UnauthorizedAccessException` as well